### DISCUSSION TITLE
Issue with RabbitMQ Deployment Using Existing PVC and Replication Setup

### DISCUSSION THREAD

**Discussion by prathi-adeptia** [rabbitmq/rabbitmq-server#12905] [Questions] [answered]
### Describe the bug

We are encountering an issue while deploying RabbitMQ with an existing PVC configuration in a Kubernetes environment.

**Steps Taken:**
Successfully deployed RabbitMQ on a single node using an existing Persistent Volume Claim (PVC). The node operates as expected.
Attempted to scale the deployment by adding replicas to enable RabbitMQ clustering.
**Observed Behavior:**
The first node continues to work without issues.
Additional replicas fail to start with errors indicating potential issues with the PVC or volume sharing.
**Configuration Details:**
RabbitMQ version: [Specify version]
**Kubernetes platform:** [e.g., AKS]
**PVC Storage Class:** [Provide details, e.g., ReadWriteOnce or ReadWriteMany]
**Helm Chart:** [Provide details, e.g., Bitnami Helm chart]
**appVersion**: 3.12.2
**helm version:** 12.0.8
Logs from failing pods: [Include specific error logs]

[rabbitmq-logs.log](https://github.com/user-attachments/files/18062895/rabbitmq-logs.log)

### Reproduction steps

1. Download and extract the RabbitMQ Bitnami helm chart.
2. Update property in values.yaml  "**existingClaim: "pv-claim-name"**"
3. Deploy RabbitMQ using helm chart 
...

### Expected behavior

**Expected Behavior:**
All replicas should start successfully and form a RabbitMQ cluster while sharing or utilizing PVCs correctly.

**Comment by mkuratczyk** [comment]
This is not about PVC sharing but about peer discovery. We can't really help since you didn't mention the version of RabbitMQ and you only share a tiny snippet of the logs from one of the nodes.

**Reply by prathi-adeptia** [reply]
Bitnami RabbtiMQ 
appVersion: 3.12.2
helm version: 12.0.8
I have configured existing pvc in "existingClaim" in values.yaml and replicas with 3.
```
 ## @param persistence.existingClaim Provide an existing PersistentVolumeClaims
  ## The value is evaluated as a template
  ## So, for example, the name can depend on .Release or .Chart
  ##
  existingClaim: ""
```
The Rabbitmq first node (xxx-xx-rabbitmq-o) started successfully. but second node (xxx-xx-rabbitmq-1) throws "exception exit: inconsistent_cluster"

**Reply by mkuratczyk** [reply, marked as answer]
RabbitMQ 3.12 is out of support and there has been significant changes in this part of the code in both 3.13 and 4.0.
If this happens with 4.0, we will investigate.

https://www.rabbitmq.com/release-information

**Reply by shanmugarajanm2020** [reply]
This happens in 3.13.6 also.

If RabbitMQ statefulset is scaled down, it can't use the existing PVCs. We had to delete those PVCs, then only rabbitMQ pods are starting.

**Reply by mkuratczyk** [reply]
RabbitMQ is not aware of PVCs - it's just folders and files as far as RabbitMQ is concerned. If you want to report a problem, please create a new discussion and provide the necessary details.

Additionally, keep in mind:
1. Scale-down is not a trivial problem. There's a reason the [cluster operator doesn't allow it at all](https://github.com/rabbitmq/cluster-operator/issues/223). It's not clear what deployment you are using, but either way - the considerations are the same. You might come up with a process that works for you, but it's hard to solve in a generic way that will just work for everyone
2. if there's a specific issue you believe to be a bug, show us how to reproduce it with RabbitMQ 4.0

**Reply by michaelklishin** [reply]
@shanmugarajanm2020 besides what's already mentioned above, [RabbitMQ 3.13 is out of community support](https://www.rabbitmq.com/release-information).
